### PR TITLE
Added forget_identity and remember_identity methods to morepath.App.

### PR DESCRIPTION
**Pull request by taschini**
This is work in progress towards #446.

### CODE REVIEW

**Comment by taschini** [comment]
The work is now pretty much done.

- The ``forget_identity`` and ``remember_identity`` methods have been added to morepath.App.
- Deprecation warnings have been added to ``morepath.settings``, ``morepath.forget_identity`` and ``morepath.remember_identity``.
- Docs and change log have been updated.

My plan is to merge this PR tomorrow, the 16th of June. Feedback is welcome, and tell me if you need more time to the review the changes.

File: doc/settings.rst [hunk 1]
```diff
@@ -48,14 +48,15 @@ Accessing a setting
...
-using the :func:`morepath.settings` function, like this::
+using the :attr:`morepath.App.settings` property::
 
-  settings().logging.logfile
+  app.settings.logging.logfile
 
-In a tween factory (see :doc:`tweens`) or a directive implementation,
-you can access a setting through the ``app`` object like this::
+Remember that the current application is also accessible from the
+reqest object::
```
Comment (href) [inline doc/settings.rst]: A little typo.

File: morepath/authentication.py [hunk 2]
```diff
@@ -69,9 +69,29 @@ def register_identity_policy_function(self, factory, dispatch, name):
             self.identity_policy = identity_policy = mapply(
                 factory,
                 settings=self.setting_registry)
-        self.reg_registry.register_function(
-            dispatch,
-            getattr(identity_policy, name))
+        import warnings
+        from functools import wraps
```
Comment (href) [inline morepath/authentication.py]: Shouldn't those go to the top of the file?

**Comment by taschini** [comment]
I incorporated your feedback, @href.

**Pull request merged**